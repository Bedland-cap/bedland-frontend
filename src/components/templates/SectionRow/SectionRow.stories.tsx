import type { Meta, StoryObj } from '@storybook/react';
import { BrowserRouter } from 'react-router-dom';
import SectionRow from './SectionRow';

const meta = {
  title: 'templates/SectionRow',
  component: SectionRow,
  tags: ['autodocs'],
  decorators: [(Story) => <BrowserRouter>{Story()}</BrowserRouter>],
} satisfies Meta<typeof SectionRow>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {
  args: {
    children: (
      <>
        <div
          style={{
            border: '1px solid white',
            height: '300px',
            width: '300px',
            color: 'white',
          }}
        >
          this is
        </div>
        <div
          style={{
            border: '1px solid white',
            height: '300px',
            width: '300px',
            color: 'white',
          }}
        >
          template
        </div>
      </>
    ),
  },
};
