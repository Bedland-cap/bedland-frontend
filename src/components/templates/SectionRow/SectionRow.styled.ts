import styled from 'styled-components';

const SectionRowLayout = styled.div`
  align-items: center;
  display: flex;
  gap: 2rem;
  height: 100%;
  justify-content: space-around;
  max-width: 1600px;
  padding: 0 2rem;
`;
export default SectionRowLayout;
