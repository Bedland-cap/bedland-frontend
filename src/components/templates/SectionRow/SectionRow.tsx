import { PropsWithChildren, useContext } from 'react';
import { ColorNames } from 'theme/theme.types';
import { ThemeContext } from 'theme/ThemeContext';
import SectionRowLayout from './SectionRow.styled';

type SectionRowProps = {
  backgroundColor?: ColorNames;
};

const defaultProps: SectionRowProps = {
  backgroundColor: 'background',
};

const SectionRow = ({
  backgroundColor = 'background',
  children,
}: PropsWithChildren<SectionRowProps>) => {
  const { palette } = useContext(ThemeContext);
  return <SectionRowLayout backgroundColor={palette[backgroundColor]}>{children}</SectionRowLayout>;
};

export default SectionRow;

SectionRow.defaultProps = defaultProps;
