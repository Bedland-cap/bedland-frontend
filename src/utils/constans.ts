export const HEADER_HEIGHT = 70;
export const FOOTER_HEIGHT = 52;
