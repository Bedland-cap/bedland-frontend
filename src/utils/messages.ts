const ERROR_MESSAGES = {
  resetPasswordPage: {
    notWithPasswordsRules:
      'Password is incorrect, please try again after reviewing the rules above',

    passwordsDoesNotMatchUp: 'Passwords do not match to each other',
  },
};

export default ERROR_MESSAGES;
