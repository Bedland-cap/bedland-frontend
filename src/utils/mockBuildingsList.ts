import { Building } from 'store/reducers/building/building_slice';

const mockBuildingsList: Building[] = [
  {
    id: '1',
    name: 'Berry',
    address: 'Berry Street 21',
    managerId: '1',
    floors: '5',
  },
  {
    id: '2',
    name: 'Miedziana',
    address: 'Miedziana Street 3',
    managerId: '1',
    floors: '7',
  },
  {
    id: '3',
    name: 'Seaport',
    address: 'Seaport Avenue 51',
    managerId: '2',
    floors: '5',
  },
  {
    id: '4',
    name: 'Desert',
    address: 'Desert Street 2X',
    managerId: '1',
    floors: '6',
  },
  {
    id: '5',
    name: 'Lakewood',
    address: 'Lakewood Avenue 57',
    managerId: '3',
    floors: '10',
  },
];

export default mockBuildingsList;
