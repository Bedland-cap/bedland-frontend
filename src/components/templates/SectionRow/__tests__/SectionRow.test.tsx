import { render } from 'test.utils';
import SectionRow from '../SectionRow';

describe('Login Page Template', () => {
  const item = () => render(<SectionRow />, {});
  it('renders correctly', () => {
    expect(item).toMatchSnapshot();
  });
});
