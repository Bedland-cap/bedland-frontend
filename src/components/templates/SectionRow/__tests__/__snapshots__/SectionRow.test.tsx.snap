// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Page Template renders correctly 1`] = `[Function]`;
